File: Python/src/card.py
import math
from random import randint


class Card:
    suits = ["heart", "diamond", "club", "spade"]
    colors = ["red", "black"]

    def __init__(self):
        self.__number = randint(1, 10)
        self.__color = self.colors[randint(0, len(self.colors)-1)]
        self.__suit = self.suits[randint(0, len(self.suits)-1)]

    # Number lottery function
    def generate_random_number(self, min, max):
        return randint(min, max)

    def __str__(self):
        return "Card {"+self.__color+", "+self.__suit+", "+str(self.__number)+"}\n"

    def draw_card(self):
        print("__________\n  " + self.__suit + "\n  " + str(self.__number) + "\n  " + self.__color + "\n__________");


    def get_number(self):
        return self.__number

    def get_color(self):
        return self.__color

    def get_suit(self):
        return self.__suit

File: C++/Card.cpp
#include "Card.h"
using namespace std;

// definw the consts values
const string Card::SUITS[4] = { "heart", "diamond", "club", "spade" };
const string Card::COLORS[2] = { "red", "black" };

Card::Card() :number(generateRandomNumber(1,10)),//Get a random number
                    suit(SUITS[generateRandomNumber(0,3)]),//Get a random color from the list
                    color(COLORS[generateRandomNumber(0,1)])//Get a random suit from the list
{}

//Number lottery function
int Card::generateRandomNumber(int min, int max)
{
    return (rand()%(max+1))+min;
}

//Describe the object
string Card::toString() const
{
    return "Card: {" + this->color + ", " + this->suit + ", " + to_string(this->number) + "}\n";;
}

//Draw a card
void Card::drawCard() const
{
    cout << "__________\n  " << this->suit << "\n  " << this->number << "\n  " << this->color << "\n__________" << std::endl;
}

//Getters
int Card::getNumber() const
{
    return this->number;
}

string Card::getSuit() const
{
    return this->suit;
}

string Card::getColor() const
{
    return this->color;
}

Card Card::operator=(const Card& card)
{
    return Card();
}